tests/sources/support.py

```diff
@@ -3,23 +3,28 @@
 # or more contributor license agreements. Licensed under the Elastic License 2.0;
 # you may not use this file except in compliance with the Elastic License 2.0.
 #
+from contextlib import asynccontextmanager
+
 from connectors.source import DEFAULT_CONFIGURATION, DataSourceConfiguration
 
 
-def create_source(klass, **extras):
+@asynccontextmanager
+async def create_source(klass, **extras):
     config = klass.get_default_configuration()
     for k, v in extras.items():
         if k in config:
             config[k].update({"value": v})
         else:
             config[k] = DEFAULT_CONFIGURATION.copy() | {"value": v}
 
-    return klass(configuration=DataSourceConfiguration(config))
+    source = klass(configuration=DataSourceConfiguration(config))
+    yield source
+    await source.close()
```

Comment:
I think you may want to put it inside `try...finally` block: [ref](https://docs.python.org/3/library/contextlib.html#contextlib.asynccontextmanager)

```suggestion
    source = klass(configuration=DataSourceConfiguration(config))
    try:
        yield source
    finally:
        await source.close()
```